mitodl/open-discussions
# Added API for listing reported content (#398)

#### What are the relevant tickets?
Fixes #398 

#### What's this PR do?
- Adds the API to list reported posts/comments
- Refactored some of our permissions to reduce duplicity and add a new `StaffOrModeratorPermission` and made sure our moderator logic is tested

#### How should this be manually tested?
- Report some posts and comments under different users
- Login as a moderator and go to `/api/v0/channels/:channel_name/reports/` and verify you see the data.
- Try for a different channel that has no reports and verify you see an empty list
- Try as a nonmoderator user and verify you get a 403
